layer/clients/model_catalog.py

```diff
@@ -1,56 +1,54 @@
 import sys
 import tempfile
+import uuid
 import warnings
 from logging import Logger
 from pathlib import Path
-from typing import TYPE_CHECKING, Optional, Tuple
+from typing import Optional, Tuple
 
-from layerapi.api.entity.model_train_pb2 import ModelTrain as PBModelTrain
-from layerapi.api.entity.model_train_status_pb2 import (
-    ModelTrainStatus as PBModelTrainStatus,
-)
-from layerapi.api.entity.model_version_pb2 import ModelVersion
-from layerapi.api.entity.source_code_environment_pb2 import SourceCodeEnvironment
-from layerapi.api.ids_pb2 import ModelTrainId, ModelVersionId
 from layerapi.api.service.modelcatalog.model_catalog_api_pb2 import (
     CompleteModelTrainRequest,
     CreateModelTrainFromVersionIdRequest,
     CreateModelTrainRequest,
     CreateModelVersionRequest,
-    CreateModelVersionResponse,
     GetModelTrainRequest,
     GetModelTrainResponse,
     GetModelTrainStorageConfigurationRequest,
     GetModelVersionRequest,
     GetModelVersionResponse,
     LoadModelTrainDataByPathRequest,
     StartModelTrainRequest,
-    StoreTrainingMetadataRequest,
     UpdateModelTrainStatusRequest,
 )
 from layerapi.api.service.modelcatalog.model_catalog_api_pb2_grpc import (
     ModelCatalogAPIStub,
 )
-from layerapi.api.value.dependency_pb2 import DependencyFile
-from layerapi.api.value.s3_path_pb2 import S3Path
 from layerapi.api.value.sha256_pb2 import Sha256
-from layerapi.api.value.source_code_pb2 import RemoteFileLocation, SourceCode
 
 from layer.cache.cache import Cache
 from layer.config import ClientConfig
 from layer.contracts.asset import AssetPath
-from layer.contracts.models import Model, ModelObject, TrainStorageConfiguration
+from layer.contracts.models import (
+    Model,
+    ModelObject,
+    ModelTrain,
+    ModelTrainStatus,
+    ModelVersion,
+    TrainStorageConfiguration,
+)
 from layer.contracts.project_full_name import ProjectFullName
 from layer.contracts.tracker import ResourceTransferState
-from layer.exceptions.exceptions import LayerClientException
+from layer.exceptions.exceptions import (
+    LayerClientException,
+    UnexpectedModelTypeException,
+)
 from layer.flavors.base import ModelRuntimeObjects
-from layer.flavors.utils import get_flavor_for_proto
+from layer.flavors.utils import get_flavor_for_model, get_flavor_for_proto
```

Comment:
not sure if you want to colocate, but both these methods are only used here now.

Reply:
I think let's leave them as is for now, and we'll likely need to move them around when we build artifacts